### PR TITLE
[LMS-374][INTEGRATION] Enroll User Page

### PR DESCRIPTION
**Pull request by FranzDelosSantos**
## Issue Link
https://framgiaph.backlog.com/board/LMS?selectedIssueKey=LMS-374&assignee=417519

## Defintion of Done
- [x] putting invalid ID must redirect to previous page (or home)
- [x] Update Breadcrumbs Text and URL
- [x] Fetch and display user from the database instead of jsonplaceholderapi

## Steps to reproduce
docker-compose up
visit url : http://localhost:3000/trainer/course/1/enrollUser

## Affected Components / Functionalities / Page
pages/trainer/course/[id]/enrollUser.tsx
shared/hooks/useEnrollUser.tsx
shared/utils/axios.ts

## Screenshots
pr14

### CODE REVIEW

File: client/src/shared/hooks/useEnrollUser.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+          if (isRequestOk(result)) {
+            const userResult = await API.get(`user/${result.data.company}`);
+            if (isRequestOk(userResult)) {
+              fetchedUsersRef.current = userResult.data.user;
+              setShowPerPage(fetchedUsersRef.current.slice(0, 10));
+            }
+          }
+        } catch (error: any) {
+          console.error(error);
+          /* alert('something went wrong'); */
```
Comment (ragg09-sunasterisk) [inline client/src/shared/hooks/useEnrollUser.ts:36]: remove comment

File: client/src/pages/trainer/course/[id]/enrollUser.tsx [hunk 2]
```diff
@@ -1,43 +1,85 @@
+/* eslint-disable @typescript-eslint/strict-boolean-expressions */
+/* eslint-disable @typescript-eslint/restrict-template-expressions */
 import React, { Fragment } from 'react';
-import { useShowPerPage } from '@/src/shared/hooks/useShowPerPage';
+
```
Comment (ragg09-sunasterisk) [inline client/src/pages/trainer/course/[id]/enrollUser.tsx:4]: remove space

File: client/src/shared/hooks/useEnrollUser.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
+/* eslint-disable @typescript-eslint/no-confusing-void-expression */
+/* eslint-disable object-shorthand */
+/* eslint-disable @typescript-eslint/restrict-template-expressions */
+import { useEffect, useRef, useState } from 'react';
+
```
Comment (ragg09-sunasterisk) [inline client/src/shared/hooks/useEnrollUser.ts:5]: remove space

File: client/src/shared/hooks/useEnrollUser.ts [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    { id: 10, text: '10' },
+    { id: 25, text: '25' },
+    { id: 50, text: '50' },
+    { id: fetchedUsersRef.current.length, text: `all (${fetchedUsersRef.current.length})` }
+  ];
+
+  const paths = [
+    {
+      text: 'Courses',
+      url: '/trainer/course'
```
Comment (ragg09-sunasterisk) [inline client/src/shared/hooks/useEnrollUser.ts:61]: this url does not exist 
```suggestion
      url: '/trainer/courses/list'
```

**Review by Marielleregner7 — APPROVED** [review]
**APPROVED**
- [ ] Enrolled users were displayed in each courses.
- [x] If course entered does not exist it will redirect to 404 page.
- [x] Show per page is working.
- [x] No ESLint error.

Comment: 0
Test time: 40 mins
Iterations: 0

Proof:
enrolluserpage

image

**Comment by ragg09-sunasterisk** [comment]
Comment: 4
Iterations: 1

**Review by ragg09-sunasterisk — APPROVED** [review]

**Pull request merged**